engines/sci/graphics/view.cpp

```diff
@@ -681,39 +655,40 @@ void GfxView::unpackCel(int16 loopNo, int16 celNo, byte *outPtr, uint32 pixelCou
 	}
 }
 
-const byte *GfxView::getBitmap(int16 loopNo, int16 celNo) {
-	loopNo = CLIP<int16>(loopNo, 0, _loopCount -1);
-	celNo = CLIP<int16>(celNo, 0, _loop[loopNo].celCount - 1);
+const SciSpan<const byte> &GfxView::getBitmap(int16 loopNo, int16 celNo) {
+	loopNo = CLIP<int16>(loopNo, 0, _loop.size() - 1);
+	celNo = CLIP<int16>(celNo, 0, _loop[loopNo].cel.size() - 1);
 	if (_loop[loopNo].cel[celNo].rawBitmap)
-		return _loop[loopNo].cel[celNo].rawBitmap;
+		return *_loop[loopNo].cel[celNo].rawBitmap;
 
 	uint16 width = _loop[loopNo].cel[celNo].width;
 	uint16 height = _loop[loopNo].cel[celNo].height;
 	// allocating memory to store cel's bitmap
-	int pixelCount = width * height;
-	_loop[loopNo].cel[celNo].rawBitmap = new byte[pixelCount];
-	byte *pBitmap = _loop[loopNo].cel[celNo].rawBitmap;
+	const uint pixelCount = width * height;
+	_loop[loopNo].cel[celNo].rawBitmap.allocate(pixelCount, _resource->getDescription() + Common::String::format(" loop %d cel %d", loopNo, celNo));
+	SciSpan<byte> rawBitmap = *_loop[loopNo].cel[celNo].rawBitmap;
```

Comment:
Once the bad implicit conversion is removed from Span, this could be changed to an explicit reinterpret_cast or C-cast to `SciSpan<byte> &`. `SpanOwner::allocate` could also be changed to return a non-const Span or raw pointer, to give write access to the caller without requiring manual allocation. Do either of these seem better? For some reason I am wary of a direct malloc here, but I will assent to it if these alternatives still look bad to you.